src/main/Extensions/ApplicationSearch/Linux/LinuxApplicationRepository.ts

```diff
@@ -0,0 +1,86 @@
+        private readonly assetPathResolver: AssetPathResolver,
+        private readonly logger: Logger,
+        private readonly settings: Settings,
+    ) {}
+
+    public async getApplications(): Promise<Application[]> {
+        try {
+            // For future support of other formats
+            const linuxFolders = this.settings.getValue<string[]>("linuxFolders");
+            const appExtensions = this.settings.getValue<string[]>("linuxFileExtensions");
+
+            this.logger.info(
+                `find -L ${linuxFolders.join(" ")} -type f -name "*${appExtensions.join('" -o -name "')}"`,
+            );
+
+            // Finds all application files
+            const promises = (
+                await this.commandlineUtility.executeCommand(
+                    `find -L ${linuxFolders.join(" ")} -type f -name "*${appExtensions.join('" -o -name "')}"`,
+                    true,
+                    true,
+                )
+            )
+                .split("\n")
+                .filter((t) => t)
+                .map(async (filePath) => {
+                    try {
+                        const config: Record<string, string> = this.iniParser.parseIniFileContent(
+                            (await this.fileSystemUtility.readFile(filePath)).toString(),
+                        )["Desktop Entry"];
+                        // Checks if app is supposed to be shown, returns null if not
+                        // https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#recognized-keys
+                        const desktopEnv = process.env["XDG_CURRENT_DESKTOP"].split(":") || ["GNOME"];
+                        if (
+                            !config ||
+                            !config["Icon"] ||
+                            config.NoDisplay ||
+                            (config.OnlyShowIn !== undefined &&
+                                !config.OnlyShowIn.split(";").some((i) => desktopEnv.includes(i))) ||
+                            (config.NotShowIn !== undefined &&
+                                config.NotShowIn.split(";").some((i) => desktopEnv.includes(i)))
+                        )
+                            return null;
+
+                        const iconImage =
+                            (await this.fileImageGenerator.getImage(filePath)) ?? this.getGenericAppIcon();
+                        return new LinuxApplication(config["Name"], filePath, iconImage);
+                    } catch (error) {
+                        this.logger.error(`Error in ApplicationSearch! Reason: ${error}`);
+                    }
+                });
+
+            const results = (await Promise.allSettled(promises))
+                .map((result) => (result.status === "fulfilled" ? result.value : null))
+                .filter((result) => result);
+
+            return results;
+        } catch (error) {
+            this.logger.error(`Error in ApplicationSearch! Error: ${error}`);
```

Comment:
Also here, no need for the exclamation mark. Also I would improve this error message that it says what failed. Here is a suggestion:

```typescript
this.logger.error(`Failed to get linux applications. Reason: ${error}`);
```